src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -206,7 +223,7 @@ private void processEmitAggregation(final EmitAggregation emitMessage) {
             for (final Request request : requests) {
                 // TODO(vkoskela): Add logging to client [MAI-89]
                 // TODO(vkoskela): Add instrumentation to client [MAI-90]
-                _pendingRequests.offer(request);
+                _pendingRequests.offer(ImmutableMap.of(request, System.currentTimeMillis()));
```

Comment:
Looks like you are trying to create a tuple. I would prefer if you just created a class with two members. It will be more maintainable and extensible.